src/components/BreedsList.vue

```diff
@@ -255,40 +257,27 @@ export default {
     },
   },
   watch: {
-    $route(newRoute, oldRoute) {
-      if (this.$route.name === "breedName") {
-        this.$router.push({
-          name: "details",
-        });
-      }
+    "$route.query.q": {
+      immediate: true,
+      handler(newRoute) {
+        if (this.$route.name === "search") {
+          this.getSearchResults(newRoute);
+        }
+      },
+    },
+    breedsList(newList, oldList) {
+      console.log(newList, oldList);
+      newList.forEach((breed) => {
+        if (breed.name.toLowerCase() === this.breedName.toLowerCase()) {
+          breed.isActive = true;
+        }
+      });
```

Comment:
- Zdej sem odstranil ves updated hook in uporabljam samo watcher za` route.query.q`, kar pomeni, da fetcham ponovno podatke glede na nov `route.query.q`,

- za `breedsList `array, kamor se fetchajo podatki. sem tudi dal pod watcher, kjer sem `breedName `props iz routerja primerjal z vsemi breedi v `breedsList `array in če se ujema, spremenim točno določenenmu breed-u object property `isActive `to `true`, kar pomeni, da če reloadam page na `details `routes-u, mi odpre točno določen breed.